extensions/notebook/src/jupyter/jupyterSessionManager.ts

```diff
@@ -171,6 +168,7 @@ export class JupyterSession implements nb.ISession {
 	private _messagesComplete: Deferred<void> = new Deferred<void>();
 
 	constructor(private sessionImpl: Session.ISession, skipSettingEnvironmentVars?: boolean, private _pythonEnvVarPath?: string) {
+		// tslint:disable-next-line:no-floating-promises This is handled by use of a deferred promise for callers to await on
```

Comment:
The correct handling is adding a .catch. It won't complain about hanging promises if you catch the errors.

Reply:
Sounds smart, can we do this?